pkg/kubelet/dockertools/manager.go

```diff
@@ -452,3 +452,40 @@ func makeCapabilites(capAdd []api.CapabilityType, capDrop []api.CapabilityType)
 	}
 	return addCaps, dropCaps
 }
+
+// PodInfraContainer returns true if the pod infra container has changed.
+func (self *DockerManager) PodInfraContainerChanged(pod *api.Pod, podInfraContainer *kubecontainer.Container) (bool, error) {
+	// Use host networking if specified and allowed.
+	networkMode := ""
+	var ports []api.ContainerPort
+
+	dockerPodInfraContainer, err := self.client.InspectContainer(string(podInfraContainer.ID))
+	if err != nil {
+		return false, err
+	}
+
+	// Check network mode.
+	if dockerPodInfraContainer.HostConfig != nil {
+		networkMode = dockerPodInfraContainer.HostConfig.NetworkMode
+	}
+	if pod.Spec.HostNetwork && networkMode != "host" {
+		return true, nil
+	} else {
+		// Docker only exports ports from the pod infra container.  Let's
+		// collect all of the relevant ports and export them.
+		for _, container := range pod.Spec.Containers {
+			ports = append(ports, container.Ports...)
+		}
+	}
+	container := &api.Container{
+		Name:  PodInfraContainerName,
+		Image: PodInfraContainerImage,
+		Ports: ports,
+	}
+
+	expectedHash := HashContainer(container)
```

Comment:
@vmarmol I have no strong opinion on this. Just because the hash approach requires less effort and won't require we parsing the result of 'docker inspect'.
Updated with a deep equal version one.